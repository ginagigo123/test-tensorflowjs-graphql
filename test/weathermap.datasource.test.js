import WeathermapAPI from '../models/weathermapDataSource'

const schema = {
  message: expect.any(String),
  cod: expect.any(String),
  count: expect.any(Number),
  list: expect.any(Array)
}

const _country = 'London'
const _weathermapAPI = new WeathermapAPI()

describe('[Weathermap.datasource]', () => {
  it('Current Weather Data Model', async () => {
    const data = await _weathermapAPI.getWeather({ country: _country })

    // Check Type
    expect(data).toMatchObject(schema)
    expect(data.cod).toBe('200')
    expect(data.count).not.toBe(0)
  })
})
