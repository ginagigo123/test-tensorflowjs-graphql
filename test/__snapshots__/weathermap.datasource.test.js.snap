// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Weathermap.datasource] Current Weather Data Model 1`] = `
Object {
  "cod": "200",
  "count": 5,
  "list": Array [
    Object {
      "clouds": Object {
        "all": 100,
      },
      "coord": Object {
        "lat": 51.5085,
        "lon": -0.1257,
      },
      "dt": 1650006552,
      "id": 2643743,
      "main": Object {
        "feels_like": 283.46,
        "humidity": 87,
        "pressure": 1027,
        "temp": 284.04,
        "temp_max": 285.29,
        "temp_min": 281.21,
      },
      "name": "London",
      "rain": null,
      "snow": null,
      "sys": Object {
        "country": "GB",
      },
      "weather": Array [
        Object {
          "description": "overcast clouds",
          "icon": "04d",
          "id": 804,
          "main": "Clouds",
        },
      ],
      "wind": Object {
        "deg": 0,
        "speed": 0.51,
      },
    },
    Object {
      "clouds": Object {
        "all": 56,
      },
      "coord": Object {
        "lat": 42.9834,
        "lon": -81.233,
      },
      "dt": 1650002956,
      "id": 6058560,
      "main": Object {
        "feels_like": 277.3,
        "humidity": 57,
        "pressure": 1001,
        "temp": 279.59,
        "temp_max": 279.95,
        "temp_min": 278.42,
      },
      "name": "London",
      "rain": null,
      "snow": null,
      "sys": Object {
        "country": "CA",
      },
      "weather": Array [
        Object {
          "description": "broken clouds",
          "icon": "04n",
          "id": 803,
          "main": "Clouds",
        },
      ],
      "wind": Object {
        "deg": 235,
        "speed": 3.13,
      },
    },
    Object {
      "clouds": Object {
        "all": 9,
      },
      "coord": Object {
        "lat": 39.8865,
        "lon": -83.4483,
      },
      "dt": 1650003020,
      "id": 4517009,
      "main": Object {
        "feels_like": 279.04,
        "humidity": 59,
        "pressure": 1019,
        "temp": 280.14,
        "temp_max": 281.68,
        "temp_min": 277.86,
      },
      "name": "London",
      "rain": null,
      "snow": null,
      "sys": Object {
        "country": "US",
      },
      "weather": Array [
        Object {
          "description": "clear sky",
          "icon": "01n",
          "id": 800,
          "main": "Clear",
        },
      ],
      "wind": Object {
        "deg": 210,
        "speed": 1.79,
      },
    },
    Object {
      "clouds": Object {
        "all": 0,
      },
      "coord": Object {
        "lat": 37.129,
        "lon": -84.0833,
      },
      "dt": 1650003020,
      "id": 4298960,
      "main": Object {
        "feels_like": 277.07,
        "humidity": 75,
        "pressure": 1022,
        "temp": 278.17,
        "temp_max": 280.25,
        "temp_min": 276.02,
      },
      "name": "London",
      "rain": null,
      "snow": null,
      "sys": Object {
        "country": "US",
      },
      "weather": Array [
        Object {
          "description": "clear sky",
          "icon": "01n",
          "id": 800,
          "main": "Clear",
        },
      ],
      "wind": Object {
        "deg": 260,
        "speed": 1.54,
      },
    },
    Object {
      "clouds": Object {
        "all": 0,
      },
      "coord": Object {
        "lat": 36.4761,
        "lon": -119.4432,
      },
      "dt": 1650003219,
      "id": 5367815,
      "main": Object {
        "feels_like": 285.76,
        "humidity": 67,
        "pressure": 1016,
        "temp": 286.61,
        "temp_max": 287.21,
        "temp_min": 284.03,
      },
      "name": "London",
      "rain": null,
      "snow": null,
      "sys": Object {
        "country": "US",
      },
      "weather": Array [
        Object {
          "description": "clear sky",
          "icon": "01n",
          "id": 800,
          "main": "Clear",
        },
      ],
      "wind": Object {
        "deg": 320,
        "speed": 4.12,
      },
    },
  ],
  "message": "accurate",
}
`;
